usr/lib/tik/modules/post/20-mig

```diff
@@ -19,16 +19,11 @@ EOF
 
 if [ "${migrate}" == 1 ]; then
     probe_partitions ${TIK_INSTALL_DEVICE} "crypto_LUKS"
-    if [ -z "${probedpart}" ]; then
-        error "encrypted partition not found"
-    fi
-    prun /usr/sbin/cryptsetup luksOpen --key-file=${tik_keyfile} ${cryptpart} aeon_root
+    [ -z "${probedpart}" ] || prun /usr/sbin/cryptsetup luksOpen --key-file=${tik_keyfile} ${cryptpart} aeon_root
 
     probe_partitions $TIK_INSTALL_DEVICE "btrfs" "/usr/lib/os-release"
 
-    if [ -z "${probedpart}" ]; then
-        error "MIGRATION FAILED: New Installation NOT FOUND"
-    fi
+    [ -n "${probedpart}" ] || error "MIGRATION FAILED: New Installation NOT FOUND"
```

Comment:
Why -n here not -z?

Reply:
We expect a partition to be found (-n), if it's not, the logical OR (||) is false after the first command and will run the second command.
If -n returns 1, the OR is already true and the second command doesn't get executed. :)